narwhals/_spark_like/namespace.py

```diff
@@ -80,12 +81,17 @@ def nth(self: Self, *column_indices: int) -> SparkLikeExpr:
         )
 
     def lit(self: Self, value: object, dtype: DType | None) -> SparkLikeExpr:
-        if dtype is not None:
-            msg = "todo"
-            raise NotImplementedError(msg)
-
         def _lit(df: SparkLikeLazyFrame) -> list[Column]:
-            return [df._F.lit(value)]
+            column = df._F.lit(value)
+            if dtype:
+                import pyspark.sql.types as pyspark_types
+
+                native_dtype = narwhals_to_native_dtype(
+                    dtype, self._version, pyspark_types
+                )
+                column = column.cast(native_dtype)
+
+            return [column.alias("literal")]
```

Comment:
We take care of the aliasing via `evaluate_output_names` below - aliasing here was causing some issues when composing with other functions.

We can just return:
```suggestion
            return [column]
```